packages/server/lib/controllers/sync.controller.ts

```diff
@@ -118,14 +118,42 @@ class SyncController {
                 return;
             }
 
-            const syncs = await getSyncs(connection, orchestrator);
+            const rawSyncs = await getSyncs(connection, orchestrator);
+
+            if (!connection.id) {
```

Comment:
note this if is not necessary, the typing is incorrect for legacy reasons, if you have a connection you have its id. You can safely suppress `connect.id!` until we fix